### PR TITLE
Search stuff

### LINKED ISSUE #27: Search
- Search box where users enter address
- when enter pressed, geocode address and search for host listings nearby (by address)

### PR DESCRIPTION
**Pull request by arjbhat**

### CODE REVIEW

File: src/app/(main)/search/page.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { SearchForm } from "@/app/(main)/search/search-form";
+import { SearchResult } from "@/app/(main)/search/search-results";
+import { usersService } from "@/services/users-service";
+import { auth } from "@clerk/nextjs";
+import { Suspense } from 'react'
+
+export default async function SearchPage() {
```
Comment (ninest) [inline src/app/(main)/search/page.tsx]: https://nextjs.org/docs/app/api-reference/file-conventions/page#searchparams-optional

The `SearchPage` component can take in `searchParams` which is an object dictionary

inside the `SearchPage`, you can do something like

```tsx
const latitude = int(searchParams.latitude)
const longitude = int(searchParams.longitude)
...
```

Use this to fetch the host listings (calling the search service) then pass them into `SearchResults`

Comment (ninest) [reply src/app/(main)/search/page.tsx]: If this is done, SearchResults will not longer have to have "use client"

**Review by ninest — CHANGES_REQUESTED** [review]

File: src/app/(main)/search/page.tsx (hunk 1 above)
Comment (ninest) [reply src/app/(main)/search/page.tsx]: @arjbhat

**Comment by ninest** [comment]
@arjbhat All you need to do know is pass the coords/qualifiers to the search service and update the hostListings variable in the search `page.tsx`, then display them in the `SearchResults` component

**Review by jamescd18 — COMMENTED** [review]
some files missing last line empty

File: src/app/(main)/search/page.tsx [hunk 2]
```diff
@@ -13,21 +14,38 @@ interface Props {
...
+  var radtheta = Math.PI * theta / 180
+  var dist = Math.sin(radlat1) * Math.sin(radlat2) + Math.cos(radlat1) * Math.cos(radlat2) * Math.cos(radtheta);
+  dist = Math.acos(dist)
+  dist = dist * 180 / Math.PI
+  dist = dist * 60 * 1.1515
+  if (unit == "K") { dist = dist * 1.609344 }
+  if (unit == "N") { dist = dist * 0.8684 }
+  return dist
+}
+
```
Comment (ninest) [inline src/app/(main)/search/page.tsx]: Doesn't mapbox have a helper function for this?

File: src/app/(main)/search/page.tsx [hunk 3]
```diff
@@ -1,17 +1,51 @@
...
   const { userId: clerkId } = auth();
   const address = clerkId ? (await usersService.getUserByClerkId(clerkId)).address : "";
 
+  const { latitude, longitude, location, qualifiers } = searchParams;
+  const hostListings = Object.keys(searchParams).length ? await hostsService.getHostListings({ qualifiers }) : [];
+  if (latitude && longitude) {
+    const lat = parseInt(latitude, 10);
+    const lon = parseInt(longitude, 10);
+
+  }
```
Comment (ninest) [inline src/app/(main)/search/page.tsx]: might wanna pass lat and lon to the search function so you don't have to do any filtering client side

**Review by ninest — APPROVED** [review]

**Pull request merged**